salt/modules/zfs.py

```diff
@@ -51,42 +62,6 @@ def _check_features():
     return res['retcode'] == 0
 
 
-def __virtual__():
-    '''
-    Makes sure that ZFS kernel module is loaded.
-    '''
-    on_freebsd = __grains__['kernel'] == 'FreeBSD'
-    on_linux = __grains__['kernel'] == 'Linux'
-    on_solaris = __grains__['kernel'] == 'SunOS' and __grains__['kernelrelease'] == '5.11'
-
-    cmd = ''
-    if on_freebsd:
-        cmd = 'kldstat -q -m zfs'
-    elif on_linux:
-        modinfo = salt.utils.path.which('modinfo')
-        if modinfo:
-            cmd = '{0} zfs'.format(modinfo)
-        else:
-            cmd = 'ls /sys/module/zfs'
-    elif on_solaris:
-        # not using salt.utils.path.which('zfs') to keep compatible with others
-        cmd = 'which zfs'
-
-    if cmd and salt.modules.cmdmod.retcode(
-        cmd, output_loglevel='quiet', ignore_retcode=True
-    ) == 0:
-        return 'zfs'
-
-    if __grains__['kernel'] == 'OpenBSD':
```

Comment:
The special exception for OpenBSD is no longer needed as we only check zfs-fuse support on linux were it makes sense.

With the code in salt.grains.zfs I do not see the service warnings anymore was the issues why the extra backout for OpenBSD was added here.